pulp_ostree/tests/functional/conftest.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import uuid
+
+from pulp_ostree.tests.functional.utils import gen_ostree_remote
+from pulpcore.client.pulp_ostree import (
+    ApiClient,
+    ContentCommitsApi,
+    ContentRefsApi,
+    RepositoriesOstreeApi,
+    RepositoriesOstreeVersionsApi,
+    RepositorySyncURL,
+    RemotesOstreeApi,
+)
+
+
+@pytest.fixture()
+def ostree_client(_api_client_set, bindings_cfg):
+    """Fixture to provide a client to Pulp API"""
+    api_client = ApiClient(bindings_cfg)
+    _api_client_set.add(api_client)
+    yield api_client
+    _api_client_set.remove(api_client)
+
+
+@pytest.fixture()
+def ostree_content_refs_api(ostree_client):
+    """Fixture that returns an instance of ContentRefsApi"""
+    return ContentRefsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RepositoriesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_remotes_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RemotesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_versions_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeVersionsApi"""
+    return RepositoriesOstreeVersionsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repository_sync_url(ostree_client):
+    """Fixture that returns an instance of RepositorySyncURL"""
+    return RepositorySyncURL(ostree_client)
+
+
+@pytest.fixture()
+def ostree_content_commits_api(ostree_client):
+    """Fixture that returns an instance of ContentCommitsApi"""
+    return ContentCommitsApi(ostree_client)
+
+
+@pytest.fixture()
+def remote(ostree_remotes_api):
+    """Fixture that creates an ostree Remote and returns the object created"""
+    body = gen_ostree_remote(depth=0, policy="immediate")
+    return ostree_remotes_api.create(body)
+
+
+@pytest.fixture()
+def random_name():
+    """Fixture that returns a dictionary with a random uuid for the `name` key"""
+    return {"name": str(uuid.uuid4())}
+
+
+@pytest.fixture()
+def repo(ostree_repositories_api, random_name):
+    """Fixture that creates an ostree Repository"""
+    return ostree_repositories_api.create(random_name)
+
+
+@pytest.fixture()
+def ostree_repository_version(
```

Comment:
Let's clearly describe what the fixture returns.

```suggestion
def synced_repo_version(
```

Furthermore, you may find it useful to call `ostree_repo_factory` here.